compaction/compaction_manager.cc

```diff
@@ -299,6 +299,7 @@ future<compaction_manager::compaction_stats_opt> compaction_manager::perform_tas
     _tasks.push_back(task);
     auto unregister_task = defer([this, task] {
         _tasks.remove(task);
+        task->_gate_holder.release();
```

Comment:
Then we should set the `_gate_holder` here, before pushing the task onto `_tasks`, instead of in the constructor.